[vulkan] Add Winograd convolutions

SS-JIA opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* **#54639 [vulkan] Add Winograd convolutions**

Differential Revision: [D27514882](https://our.internmc.facebook.com/intern/diff/D27514882)

Commit:
[vulkan] Add Winograd convolutions

[ghstack-poisoned]

Commit:
Update on "[vulkan] Add Winograd convolutions"




[ghstack-poisoned]

Commit:
Update on "[vulkan] Add Winograd convolutions"




[ghstack-poisoned]

Commit:
Update on "[vulkan] Add Winograd convolutions"




[ghstack-poisoned]

Commit:
Update on "[vulkan] Add Winograd convolutions"




[ghstack-poisoned]

Commit:
Update on "[vulkan] Add Winograd convolutions"




[ghstack-poisoned]

Commit:
Update on "[vulkan] Add Winograd convolutions"




[ghstack-poisoned]

Commit:
Update on "[vulkan] Add Winograd convolutions"




[ghstack-poisoned]

Commit:
Update on "[vulkan] Add Winograd convolutions"




[ghstack-poisoned]

Commit:
Update on "[vulkan] Add Winograd convolutions"




[ghstack-poisoned]

IvanKobzarev commented on aten/src/ATen/test/vulkan_api_test.cpp:
```diff
@@ -606,6 +606,78 @@ TEST(VulkanAPITest, conv2d_pw) {
+  } weights {13, input.channels, 3, 3};
+
+  const auto input_cpu = at::rand(input.size(), at::device(at::kCPU).dtype(at::kFloat));
+  const auto weights_cpu = at::rand(weights.size(), at::device(at::kCPU).dtype(at::kFloat))*2;
```
why '*2' in the end?

SS-JIA replied:
Same as above

IvanKobzarev commented on aten/src/ATen/test/vulkan_api_test.cpp:
```diff
@@ -430,11 +430,11 @@ TEST(VulkanAPITest, conv2d) {
-  const auto bias_cpu = at::randn({weights.output_channels}, at::device(at::kCPU).dtype(at::kFloat));
+  const auto input_cpu = at::randn(input.size(), at::device(at::kCPU).dtype(at::kFloat))*5;
+  const auto weights_cpu = at::randn(weights.size(), at::device(at::kCPU).dtype(at::kFloat))*5;
+  const auto bias_cpu = at::randn({weights.output_channels}, at::device(at::kCPU).dtype(at::kFloat))*5;
```
why *5, testing ?

SS-JIA replied:
ah, I added these during testing and forgot to remove them, same for the one below. Good eye!

Commit:
Update on "[vulkan] Add Winograd convolutions"


Differential Revision: [D27514882](https://our.internmc.facebook.com/intern/diff/D27514882)

[ghstack-poisoned]
